### PR TITLE
Added `tab` focus mode to the Breadcrumb

### PR DESCRIPTION
**Pull request by ValentinaKozlova**
After discussion with the design team and Sarah we have to change focus behavior.
## Previous Behavior
- default mode was `arrow`
- no `tab` navigation

## New Behavior
- `tab` navigation became the default one
- added cypress tests

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/98c157fc502beffdee224d0b8274d700ef5ea1d8'>98c157fc502beffdee224d0b8274d700ef5ea1d8</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by ValentinaKozlova** [comment]
> cypress is not properly setup
> 
> the fix should be running `yarn nx workspace-generators migrate-converged-pkg --name @fluentui/react-breadcrumb`, but it contains [bug](https://github.com/microsoft/fluentui/issues/27223) which wont update your project thus you need to do it manually:
> 
> 1. create `tsconfig.cy.json` in your project root
> 
> ```json
> {
>   "extends": "./tsconfig.json",
>   "compilerOptions": {
>     "isolatedModules": false,
>     "types": ["node", "cypress", "cypress-storybook/cypress", "cypress-real-events"],
>     "lib": ["ES2019", "dom"]
>   },
>   "include": ["**/*.cy.ts", "**/*.cy.tsx"]
> }
> ```
> 
> 2. update `tsconfig.json.lib`:
> 
> ```diff
>  "exclude": [
>     "./src/testing/**",
>     "**/*.spec.ts",
>     "**/*.spec.tsx",
>     "**/*.test.ts",
>     "**/*.test.tsx",
>     "**/*.stories.ts",
>     "**/*.stories.tsx",
> +    "**/*.cy.ts",
> +    "**/*.cy.tsx"
>   ],
> ```
> 
> 2. update `tsconfig.lib`:
> 
> ```diff
> "references": [
>     {
>       "path": "./tsconfig.lib.json"
>     },
>     {
>       "path": "./tsconfig.spec.json"
>     },
>     {
>       "path": "./.storybook/tsconfig.json"
>     },
> +    {
> +      "path": "./tsconfig.cy.json"
> +    }
>   ]
> ```

Thank you! :)

**Review by marcosmoura — APPROVED** [review]

File: packages/react-components/react-breadcrumb/src/components/BreadcrumbLink/useBreadcrumbLink.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export const useBreadcrumbLink_unstable = (
     root: getNativeElementProps('a', {
       'aria-current': current ? props['aria-current'] ?? 'page' : undefined,
       ref,
+      tabIndex: focusMode === 'off' ? -1 : undefined,
```
Comment (ling1726) [inline packages/react-components/react-breadcrumb/src/components/BreadcrumbLink/useBreadcrumbLink.ts:33]: does it make sense for breadcrumb links to not be focusable? they should not be rendered semantically as a `a` tag then

Comment (ValentinaKozlova) [reply packages/react-components/react-breadcrumb/src/components/BreadcrumbLink/useBreadcrumbLink.ts:33]: We discussed with Sarah that it won't hurt to turn it off. But we talked about non-interactive controls and in our case if people expect non-interactive Breadcrumb they can use only BreacrumbItem. I'll remove `off` focus mode

Comment (ValentinaKozlova) [reply packages/react-components/react-breadcrumb/src/components/BreadcrumbLink/useBreadcrumbLink.ts]: resolving as this is fixed

**Pull request merged**